modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraAssignmentsRepository.java

```diff
@@ -556,6 +552,196 @@ public Assignment assignUser(Assignment assignment, Experiment experiment, Date
+    }
+
+    /**
+     * Submit tasks for each assignment to increment/decrement counts
+     *
+     * @param assignments
+     * @param date
+     */
+    private void incrementCounts(List<Pair<Experiment, Assignment>> assignments, Date date) {
+        boolean countUp = true;
+        assignments.forEach(pair -> {
+            assignmentsCountExecutor.execute(new AssignmentCountEnvelope(this, experimentRepository, dbRepository, pair.getLeft(), pair.getRight(), countUp, eventLog, date, assignUserToExport, assignBucketCount));
+        });
+        logger.debug("Finished assignmentsCountExecutor");
+    }
+
+    /**
+     * Make entries in user_bucket_index table
+     *
+     * @param assignments
+     */
+    private void indexUserToBucket(List<Pair<Experiment, Assignment>> assignments) {
+        final List<ResultSetFuture> rFutures = new ArrayList<>();
+        assignments.forEach(pair -> {
+            rFutures.add(asyncIndexUserToBucket(pair.getRight()));
+        });
+
+        rFutures.forEach(ResultSetFuture::getUninterruptibly);
+        logger.debug("Finished asyncIndexUserToBucket");
+    }
+
+    /**
+     * Make entries in experiment_user_index table
+     *
+     * @param assignments
+     */
+    private void indexExperimentsToUser(List<Pair<Experiment, Assignment>> assignments) {
+        try {
+            Session session = driver.getSession();
+            final BatchStatement batchStatement = new BatchStatement(BatchStatement.Type.UNLOGGED);
+            assignments.forEach(pair -> {
+                Assignment assignment = pair.getRight();
+                logger.debug("assignment={}", assignment);
+                BoundStatement bs;
+                if (isNull(assignment.getBucketLabel())) {
+                    bs = experimentUserIndexAccessor.insertBoundStatement(assignment.getUserID().toString(), assignment.getContext().toString(), assignment.getApplicationName().toString(), assignment.getExperimentID().getRawID());
+                } else {
+                    bs = experimentUserIndexAccessor.insertBoundStatement(assignment.getUserID().toString(), assignment.getContext().toString(), assignment.getApplicationName().toString(), assignment.getExperimentID().getRawID(), assignment.getBucketLabel().toString());
+                }
+                batchStatement.add(bs);
+            });
+            session.execute(batchStatement);
+            logger.debug("Finished experiment_user_index");
+        } catch(Exception e) {
+            logger.error("Error occurred while adding data in to experiment_user_index", e);
+        }
+    }
+
+    /**
+     *
+     * Index user to bucket in asynchronous way
+     *
+     * @param assignment
+     * @return
+     */
+    ResultSetFuture asyncIndexUserToBucket(Assignment assignment) {
+        ResultSetFuture resultSetFuture = null;
+        try{
+            if(isNull(assignment.getBucketLabel())) {
+                resultSetFuture = userBucketIndexAccessor.asyncInsertBy(
```

Comment:
Are we deprecating userbucket index - https://trello.com/c/oGNYePZa/1152-deprecate-user-bucket-index-table-and-use-user-bucket-count-table-instead ?